plugin/pulp/plugin/tasking.py

```diff
@@ -25,15 +26,13 @@ def append_non_fatal_error(cls, error):
         attribute on the :class: `~pulp.app.models.Task` model.
 
         :param error: The non fatal error to be appended.
-        :type error: dict
+        :type error: Exception
 
         :raises :class: `pulp.app.models.Task.DoesNotExist`: If not currently running inside a
                                                              task.
         """
         task_id = get_current_task_id()
         task_obj = Task.objects.get(id=task_id)
-        if not isinstance(task_obj.non_fatal_errors, list):
-            task_obj.non_fatal_errors = [error]
-        else:
-            task_obj.non_fata_errors.append(error)
+        serialized_error = exception_to_dict(error, None)
```

Comment:
Since we are not planning to store tracebacks for non fatal errors, that is the way I will implement it. thanks for the suggestion.